eclipse-collections/src/main/java/org/eclipse/collections/impl/list/primitive/LongInterval.java

```diff
@@ -0,0 +1,1106 @@
+                    + LongInterval.rangeSize(from, to, stepBy)
+                    + " exceeds max size() of "
+                    + Integer.MAX_VALUE);
+        }
+        return new LongInterval(from, to, stepBy);
+    }
+
+    private static long rangeSize(long from, long to, long step)
+    {
+        return (to - from) / step + 1L;
+    }
+
+    /**
+     * Returns true if the LongInterval contains all of the specified long values.
+     */
+    @Override
+    public boolean containsAll(long... values)
+    {
+        for (long value : values)
+        {
+            if (!this.contains(value))
+            {
+                return false;
+            }
+        }
+        return true;
+    }
+
+    @Override
+    public boolean containsAll(LongIterable source)
+    {
+        for (LongIterator iterator = source.longIterator(); iterator.hasNext(); )
+        {
+            if (!this.contains(iterator.next()))
+            {
+                return false;
+            }
+        }
+        return true;
+    }
+
+    /**
+     * Returns true if the LongInterval contains none of the specified long values.
+     */
+    public boolean containsNone(long... values)
+    {
+        for (long value : values)
+        {
+            if (this.contains(value))
+            {
+                return false;
+            }
+        }
+        return true;
+    }
+
+    /**
+     * Returns true if the LongInterval contains the specified long value.
+     */
+    @Override
+    public boolean contains(long value)
+    {
+        return this.isWithinBoundaries(value) && (value - this.from) % this.step == 0L;
+    }
+
+    private boolean isWithinBoundaries(long value)
+    {
+        return this.step > 0L && this.from <= value && value <= this.to
+                || this.step < 0L && this.to <= value && value <= this.from;
+    }
+
+    @Override
+    public void forEachWithIndex(LongIntProcedure procedure)
+    {
+        int index = 0;
+        if (this.from <= this.to)
+        {
+            for (long l = this.from; l <= this.to; l += this.step)
+            {
+                procedure.value(l, index++);
+            }
+        }
+        else
+        {
+            for (long l = this.from; l >= this.to; l += this.step)
+            {
+                procedure.value(l, index++);
+            }
+        }
+    }
+
+    @Override
+    public void forEach(LongProcedure procedure)
+    {
+        this.each(procedure);
+    }
+
+    @Override
+    public void each(LongProcedure procedure)
+    {
+        if (this.from <= this.to)
+        {
+            for (long l = this.from; l <= this.to; l += this.step)
+            {
+                procedure.value(l);
+            }
+        }
+        else
+        {
+            for (long l = this.from; l >= this.to; l += this.step)
+            {
+                procedure.value(l);
+            }
+        }
+    }
+
+    @Override
+    public int count(LongPredicate predicate)
+    {
+        int count = 0;
```

Comment:
this will be an issue for LongInterval.fromTo(0, Long.MAX_VALUE) or Long.MIN_VALUE, 0) and alwaysTrue.
I do not know what the solution is since we need to override the count() method, but may be our count() should return a long for LongInterval as an overloaded method?